src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java

```diff
@@ -0,0 +1,316 @@
+            var batchItems = new BatchItems();
+
+            for (var currentEod : allPendingSourceToArmEntities) {
+
+                var batchItem = new BatchItem();
+
+                try {
+                    ExternalObjectDirectoryEntity armEod;
+                    if (isEqual(currentEod.getExternalLocationType(), EodHelper.armLocation())) {
+                        armEod = updateArmEodToArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile);
+                        if (armEod == null) {
+                            continue;
+                        }
+                    } else {
+                        armEod = createArmEodWithArmIngestionStatus(currentEod, batchItem, batchItems, archiveRecordsFile);
+                    }
+
+                    String rawFilename = generateFilename(armEod);
+                    if (shouldPushRawDataToArm(batchItem)) {
+                        pushRawDataAndCreateArchiveRecordIfSuccess(batchItem, rawFilename);
+                    } else if (shouldAddEntryToManifestFile(batchItem)) {
+                        var archiveRecord = archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename);
+                        batchItem.setArchiveRecord(archiveRecord);
+                    }
+                } catch (Exception e) {
+                    log.error("Unable to batch push EOD {} to ARM", currentEod.getId(), e);
+                    recoverByUpdatingEodToFailedArmStatus(batchItem);
+                }
+            }
+
+            try {
+                if (!batchItems.getSuccessful().isEmpty()) {
+                    writeManifestFile(batchItems, archiveRecordsFile);
+                    copyMetadataToArm(archiveRecordsFile);
+                }
+            } catch (Exception e) {
+                log.error("Error during generation of batch manifest file {}", archiveRecordsFile.getName(), e);
+                batchItems.getSuccessful().forEach(this::recoverByUpdatingEodToFailedArmStatus);
+                return;
+            }
+
+            for (var batchItem : batchItems.getSuccessful()) {
+                updateExternalObjectDirectoryStatus(batchItem.getArmEod(), EodHelper.armDropZoneStatus());
+            }
+        }
+
+        log.info("Finished running ARM Batch Push processing at: {}", OffsetDateTime.now());
+    }
+
+    private ExternalLocationTypeEntity getEodSourceLocation() {
+        var armClient = armDataManagementConfiguration.getArmClient();
+        if (armClient.equalsIgnoreCase("darts")) {
+            return EodHelper.unstructuredLocation();
+        } else if (armClient.equalsIgnoreCase("dets")) {
+            return EodHelper.detsLocation();
+        } else {
+            throw new DartsException(String.format("Invalid arm client '%s'", armClient));
+        }
+    }
+
+    private List<ExternalObjectDirectoryEntity> getArmExternalObjectDirectoryEntities(int batchSize) {
```

Comment:
So this method will prioritise new EOD records that have not been attempted to be pushed to ARM over those that have already been attempted?

Reply:
this is a valid point. But the job eventually will push them all anyway, right. If we believe it's important prioritising both types of eods in each run, I will make the change. Might not be an easy one though

Reply:
I think you misunderstood what I mean't. If you swap them around so you get batch from the existing records first and then get the remainder of the new records if there is a remainder. @hemantasharma1129 what are your thoughts on this?

Reply:
ok I see, you're suggesting to just swap them. Ok sure happy to do that. Let's hear from Hemanta first